src/vs/workbench/contrib/preferences/browser/settingsTree.ts

```diff
@@ -353,27 +354,78 @@ export function resolveConfiguredUntrustedSettings(groups: ISettingsGroup[], tar
 	return [...allSettings].filter(setting => setting.restricted && inspectSetting(setting.key, target, configurationService).isConfigured);
 }
 
-export function resolveExtensionsSettings(groups: ISettingsGroup[]): ITOCEntry<ISetting> {
-	const settingsGroupToEntry = (group: ISettingsGroup) => {
+export async function resolveExtensionsSettings(extensionService: IExtensionService, groups: ISettingsGroup[]): Promise<ITOCEntry<ISetting>> {
+	const extGroupTree = new Map<string, ITOCEntry<ISetting>>();
+	const addEntryToTree = (extensionId: string, extensionName: string, childEntry: ITOCEntry<ISetting>) => {
+		if (!extGroupTree.has(extensionId)) {
+			const rootEntry = {
+				id: extensionId,
+				label: extensionName,
+				children: []
+			};
+			extGroupTree.set(extensionId, rootEntry);
+		}
+		extGroupTree.get(extensionId)!.children!.push(childEntry);
+	};
+	const processGroupEntry = async (group: ISettingsGroup) => {
 		const flatSettings = arrays.flatten(
 			group.sections.map(section => section.settings));
 
-		return {
+		const extensionId = group.extensionInfo!.id;
+		const extension = await extensionService.getExtension(extensionId);
+		const extensionName = extension!.displayName ?? extension!.name;
+
+		let sectionOrder = 0;
+		const extensionConfiguration = extension?.contributes?.configuration;
+		if (Array.isArray(extensionConfiguration)) {
+			const currentSection = extensionConfiguration.find((config) => (config as any)?.id === group.id);
+			if (currentSection) {
+				sectionOrder = (currentSection as any)?.order ?? 0;
+			}
+		}
+
+		const childEntry = {
 			id: group.id,
 			label: group.title,
+			order: sectionOrder,
 			settings: flatSettings
 		};
+		addEntryToTree(extensionId, extensionName, childEntry);
 	};
 
-	const extGroups = groups
+	const processPromises = groups
 		.sort((a, b) => a.title.localeCompare(b.title))
-		.map(g => settingsGroupToEntry(g));
+		.map(g => processGroupEntry(g));
+
+	return Promise.all(processPromises).then(() => {
+		const extGroups: ITOCEntry<ISetting>[] = [];
+		for (const value of extGroupTree.values()) {
+			if (value.children!.length === 1) {
+				// push a flattened setting
```

Comment:
This is fine but gitlens will lose the description that it puts in the section title. We can ask Eric how annoying that is, but if he adopts the section view by splitting his settings into sections, then he can keep that description on the top section.

Reply:
So basically if you haven't already, talk to him to see whether he would want to adopt it right away, or if he's ok with losing the extra description until then, because we shouldn't merge this and surprise him by losing that message

Reply:
I changed the flattening case to use `value.children![0].label` now instead, which uses the same labels as in Insiders.